token/wallet.tact

```diff
@@ -2,164 +2,116 @@
+            self.owner
+                .notify_transfer(
+                    transfer.query_id,
+                    transfer.amount,
+                    msg.from,
+                    payload,
+                )
+                .send(ihr_disabled: true, pay_transfer_fee_separately: true);
+        }
+
+        if (!transfer.response_address.is_none() && msg_value > 0) {
+            transfer
+                .response_address
+                .excesses(msg.query_id)
+                .send(
+                    money: msg.amount,
+                    ihr_disabled: true,
+                    ignore_action_phase_errors: true
+                );
+        }
     }
-}
 
-fn burn_tokens(
-    @mut state: WalletState, 
-    msg: Burn, 
-    sender: ContractAddress, 
-    msg_value: Coins
-    fwd_fee: Coins, 
-) {
-    verify(state.owner == sender, "sender must be owner");
-    verify(state.balance - msg.amount >= 0, "not enough balance");
-
-    let required_input = fwd_fee + 2 * GAS_CONSUMPTION;
-    verify(msg_value > required_input, "not enough value");
-
-    state.balance -= msg.amount;
-
-    RequestBuilder.new()
-        .ihr_disabled()
-        .can_be_bounced()
-        .with_coins(CarryMessageInput{})
-        .body_as_ref[InternalMsgBody](BurnNotification {
-            query_id: msg.query_id,
-            amount: msg.amount,
-            from: state.owner,
-            response_address: msg.response_destination,
-        })
-        .send_to(state.master);
-}
-
-fn on_bounce(
-    @mut state: WalletState,
-    body: Slice
-) {
-    let op = body.load_uint(32);
-    verify(op == InternalTransfer || op == BurnNotification, "invalid bounce operation");
+    internal burn_tokens#595f07bc(query_id: Uint64, burn_amount: JettonAmount, response_destination: ContractAddress) {
+        verify(self.owner == msg.sender, "sender must be owner");
+        verify(burn_amount <= self.balance, "insufficient balance");
 
-    let query_id = body.load_uint(64);
-    let jetton_amount = body.load_coins();
-    state.balance += jetton_amount;
-}
-
-fn recv_internal(my_balance: Coins, msg_value: Coins, in_msg_full: Cell) {
-    if (in_msg_body.slice_empty()) { // ignore empty messages
-        return;
-    }
-    let msg = InternalMessage[InternalMsgBody].deserialize(in_msg_full);
-    if (msg.info.bounced) {
-        on_bounce(msg.body);
-        return;
-    }
+        let required_input = msg.fwd_fee + 2 * GAS_CONSUMPTION;
+        verify(msg.amount < required_input, "not enough value");
 
-    let mut state = WalletState.deserialize(Global.get_state());
+        self.balance -= burn_amount;
 
-    switch (msg.body) {
-        Transfer => send_tokens(state, msg.body, msg.info.sender, msg_value, msg.info.fwd_fee);
-        InternalTransfer => receive_tokens(state, msg.body, msg.info.sender, my_balance, msg.info.fwd_fee, msg_value);
-        Burn => burn_tokens(state, msg.body, msg.info.sender, msg_value, msg.info.fwd_fee);
+        self.master
```

Comment:
Based on our discussions this would look something like this:

```
send(InboundCoins) self.master.burn_notification(
   query_id, 
   amount: burn_amount, 
   from: self.owner, 
   response_address,
   msg.flags.bounceable: true)
```

Reply:
Updated in [d6ca35f](https://github.com/tact-lang/tact-bikeshed/pull/5/commits/d6ca35fb23d43bb86e10c754a7f34e1e3668d01f)